### PR TITLE
Allowed memory size of 1073741824 bytes exhausted

### PR DESCRIPTION
**Pull request by ehovel**
add CURLOPT_WRITEFUNCTION  when i need to download big file

### CODE REVIEW

**Comment by ehovel** [comment]
maybe we should add file lock

File: src/org/apache/hadoop/tools/Curl.php [hunk 1]
```diff
@@ -133,6 +143,17 @@ private function _exec($options, $returnInfo=false) {
 		} else {
 			$options[CURLOPT_HTTPHEADER] = array_merge($options[CURLOPT_HTTPHEADER], array('Expect: '));
 		}
+		if (isset($options['CURLOPT_WRITEFUNCTION_PATH'])) {
+            $localPath = $options['CURLOPT_WRITEFUNCTION_PATH'];
+            $options[CURLOPT_WRITEFUNCTION] = function ($ch, $string) use ($localPath) {
+                $fp = fopen($localPath,'a');
+                $length = fwrite($fp, $string);
+                fclose($fp);
```
Comment (xaviered) [inline src/org/apache/hadoop/tools/Curl.php]: Do we also want to do `fflush($fp)`?

**Review by xaviered — CHANGES_REQUESTED** [review]
Please up the version https://github.com/xaviered/php-WebHDFS/blob/master/composer.json#L6 to `1.0.5`

File: src/org/apache/hadoop/tools/Curl.php (hunk 1 above)
Comment (xaviered) [reply src/org/apache/hadoop/tools/Curl.php]: Also, throw exceptions for `fopen` and `fwrite`

**Comment by xaviered** [comment]
Please rebase, as I just sent over a big change to use PSR-2 coding style standards

**Comment by ehovel** [comment]
updated

**Review by ehovel — COMMENTED** [review]
updated version

**Comment by xaviered** [comment]
@ehovel sorry for the very late follow up on this. I've added a couple of memory fixes too, and I've added your fix to a new version. You can get it at https://packagist.org/packages/simpleenergy/php-webhdfs#1.0.6

**Pull request merged**